Sources/Purchasing/TrialOrIntroPriceEligibilityChecker.swift

```diff
@@ -82,7 +82,7 @@ class TrialOrIntroPriceEligibilityChecker {
 
     func sk1CheckEligibility(_ productIdentifiers: [String],
                              completion: @escaping ReceiveIntroEligibilityBlock) {
-        receiptFetcher.receiptData(refreshPolicy: .never) { data in
+        receiptFetcher.receiptData(refreshPolicy: .onlyIfEmpty) { data in
```

Comment:
cc @aboedo any reason why this should be `.never`?

Reply:
yeah, sadly. 
refreshing here will likely prompt the user for their credentials, and intro eligibility should be triggered programmatically. 

In theory, this should only be a problem in sandbox, but I have a vague recollection of having moved to `never` after this caused a lot of customer confusion and headaches

Reply:
that makes total sense. 

Also, would you mind adding what I explained as a comment? Your question about why this should be `.never` is very valid and there's no reason maintainers should know the explanation before seeing this code, I should have added a comment myself when adding the `.never` 🤦